certbot-apache/certbot_apache/dualparser.py

```diff
@@ -0,0 +1,266 @@
+""" Tests for ParserNode interface """
+from certbot_apache import assertions
+from certbot_apache import augeasparser
+
+
+class DualNodeBase(assertions.InterfaceAssertions):
+    """ Dual parser interface for in development testing. This is used as the
+    base class for dual parser interface classes. This class handles runtime
+    attribute value assertions."""
+
+    def save(self, msg):
+        """ Call save for both parsers """
+        self.primary.save(msg)
+        self.secondary.save(msg)
```

Comment:
I think this should probably come in (much) later PRs, but just to flag it now, before we ship this code, I think we need to figure out more of the details of our migration plan and implement things like good reporting and error handling around all of the calls here.